packages/eas-cli/src/commands/update/index.ts

```diff
@@ -509,9 +509,9 @@ export default class UpdatePublish extends EasCommand {
     const nonInteractive = flags['non-interactive'] ?? false;
 
     const { auto, branch: branchName, channel: channelName, message: updateMessage } = flags;
-    if (nonInteractive && !auto && !(branchName && channelName && updateMessage)) {
+    if (nonInteractive && !auto && !(updateMessage && (branchName || channelName))) {
       Errors.error(
-        '--auto or both --channel or --branch and --message are required when updating in non-interactive mode',
+        '--auto or --message and either --branch or --channel are required when updating in non-interactive mode',
```

Comment:
This is an improvement, but I think this error is still not clear. I sounds to me that (auto or message) and (channel  or branch) is required.

Reply:
Open for other suggestions as well :)

Reply:
```suggestion
        '--message and either --branch or --channel are required when updating in non-interactive mode unless --auto flag is specified ',
```
maybe sth like this?

Reply:
Let's combine them:

```suggestion
      '--branch and --message, or --channel and --message are required when updating in non-interactive mode unless --auto is specified',
```